### PR TITLE
New: Improved Discover recommendations

### LINKED ISSUE #5590: Improve Discovery Recommendations Query
**Is your feature request related to a problem? Please describe.**
Discovery should show most recommended movies instead of just recommendations from the last 10 added movies

**Additional context**
SQL as discussed on Discord

### PR DESCRIPTION
**Pull request by bakerboy448**
New: Movies existing in Library no longer show as a Discover Recommendation
New: Improved Discover recommendations
#### Database Migration
NO

#### Description
Improved Discover Recommendations so that it returns the 100 most frequently recommended movies from the user's library. This is much better than grabbing the literal `top` 100 recommendations from the table. This will also exclude existing movies in the library from being recommended...that was kinda dumb.
#### Todos
- [X] Tests (None)
- [X] Translation Keys (None)
- [X] Wiki Updates (None; Not yet built out)
- [ ] Test With Large Library @Qstick <<<

### Testing
967 Movies
0 Lists
Baseline SQL: 5ms
UI Load (no cache):  2.68s
New SQL: 24ms
UI Load (no cache) : 2.33s

#### Issues Fixed or Closed by this PR

* Fixes #5590

### CODE REVIEW

File: src/NzbDrone.Core/Movies/MovieRepository.cs [hunk 1]
```diff
@@ -313,19 +313,20 @@ public List<int> AllMovieTmdbIds()
 
         public List<int> GetRecommendations()
         {
-            var recommendations = new List<List<int>>();
+            var recommendations = new List<int>();
             var tmdbIds = AllMovieTmdbIds();
```
Comment (Qstick) [inline src/NzbDrone.Core/Movies/MovieRepository.cs:317]: Blitz this, its not used anymore and just slows things down by making another call.

File: src/NzbDrone.Core/Movies/MovieRepository.cs [hunk 2]
```diff
@@ -313,19 +313,20 @@ public List<int> AllMovieTmdbIds()
...
 
             using (var conn = _database.OpenConnection())
             {
-                recommendations =  conn.Query<List<int>>("SELECT Recommendations FROM Movies ORDER BY id DESC LIMIT 100").ToList();
+                recommendations = conn.Query<int>(@"
+                     SELECT Recommendations FROM
+                    (SELECT Recommendations, count(*) AS recCount FROM
+                    (SELECT CAST(j.value AS INT) AS Recommendations FROM Movies CROSS JOIN json_each(Movies.recommendations) AS j
+                    WHERE Recommendations NOT IN (SELECT TmdbId FROM Movies))
+                    GROUP BY Recommendations ORDER BY recCount desc LIMIT 100);").ToList();
```
Comment (Qstick) [inline src/NzbDrone.Core/Movies/MovieRepository.cs:326]: SQL can be simplified, also the exisitng check doesn't work as you have the alias named the same as a column in movies table. Try this

```
SELECT Rec FROM
(SELECT CAST(j.value AS INT) AS Rec FROM Movies CROSS JOIN json_each(Movies.Recommendations) AS j
WHERE Rec NOT IN (SELECT TmdbId FROM Movies))
GROUP BY Rec ORDER BY count(*) DESC LIMIT 100;
```

File: src/NzbDrone.Core/Movies/MovieService.cs [hunk 3]
```diff
@@ -286,7 +286,7 @@ public void RemoveAddOptions(Movie movie)
         public void SetFileId(Movie movie, MovieFile movieFile)
         {
             _movieRepository.SetFileId(movieFile.Id, movie.Id);
-            _logger.Info("Linking [{0}] > [{1}]", movieFile.RelativePath, movie);
+            _logger.Info("Assigning [{0}] to movie [{1}]", movieFile.RelativePath, movie);
```
Comment (Qstick) [inline src/NzbDrone.Core/Movies/MovieService.cs:289]: `Assigning file [{0}]...`

File: src/NzbDrone.Core/Movies/MovieService.cs [hunk 4]
```diff
@@ -386,7 +386,7 @@ public void Handle(MovieFileAddedEvent message)
             _movieRepository.Update(movie);
 
             //_movieRepository.SetFileId(message.MovieFile.Id, message.MovieFile.Movie.Value.Id);
-            _logger.Info("Linking [{0}] > [{1}]", message.MovieFile.RelativePath, message.MovieFile.Movie);
+            _logger.Info("Assigning [{0}] to movie [{1}]", message.MovieFile.RelativePath, message.MovieFile.Movie);
```
Comment (Qstick) [inline src/NzbDrone.Core/Movies/MovieService.cs:389]: `Assigning file [{0}]...`

**Review by Qstick — COMMENTED** [review]
A few comments, This seems to be a bit slower but should be fairly even when you remove the AllMovieTmdbIds call.

File: src/NzbDrone.Core/Movies/MovieRepository.cs (hunk 2 above)
Comment (bakerboy448) [reply src/NzbDrone.Core/Movies/MovieRepository.cs:326]: tested (in SQL browser only)  nice catch!  seems to work!

**Review by Qstick — APPROVED** [review]

**Pull request merged**